yaksh/views.py

```diff
@@ -3237,3 +3237,51 @@ def download_course_progress(request, course_id):
     for student in stud_details:
         writer.writerow(student)
     return response
+
+
+@login_required
+@email_verified
+def download_course_yaml(request, course_id):
+    user = request.user
+    course = Course.objects.get(id=course_id)
+    if not is_moderator(user):
+        raise Http404('You are not allowed to download this course !')
+    if not course.is_creator(user):
```

Comment:
Shouldn't the course teacher be allowed to download the course?

Reply:
Currently they are not, because ownership of uploaded courses is complicating the logic. We'll see the usage of the feature and revisit if required